catroid/src/main/java/org/catrobat/catroid/io/SoundManager.java

```diff
@@ -148,6 +164,27 @@ public synchronized void stopAllSounds() {
 		}
 	}
 
+	public Map<String, Integer> getPlayingSoundDurationMap() {
+		List<Integer> positionList =
+				SoundManager.getInstance().getCurrentPositionOfSounds();
+		Map<String, Integer> soundsDurationMap = new HashMap<>();
+		for (String sound : soundFiles) {
+			int position = positionList.get(soundFiles.indexOf(sound));
```

Comment:
This seems a bit too complicated for this task. Instead of two separate lists of sound files and media Players. You could map one sound file to one MediaPlayer from the beginning (in play sound).
For further processing see https://mkyong.com/java8/java-8-filter-a-map-examples/
 map<String, MediaPlayer> => filter => map => collect = map<string, Integer>

Reply:
Good Points, thanks!
Unfortunatly using stream is only possible starting with API 24 and as per my IDE is the current min 21.

Reply:
Ok, then its probably fine as it is. You can decide if you want to refactor this or leave it like that.